### PR TITLE
OTA-1232: hypershift z-stream downgrade

### PR DESCRIPTION
**Pull request by LiangquanLi930**

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-4.15-upgrade-from-stable-4.15-aws-ipi-ovn-hypershift-inplace-rollback-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-4.15-upgrade-from-stable-4.15-aws-ipi-ovn-hypershift-inplace-rollback-f7

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly-4.15-upgrade-from-stable-4.15.yaml [hunk 1]
```diff
@@ -56,6 +56,12 @@ releases:
       version_bounds:
         lower: 4.15.0-0
         upper: 4.16.0-0
+  oldest-supported-z-rollback:
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly-4.15-upgrade-from-stable-4.15.yaml:59]: We have dedicated ci yaml for rollback ci:
```
[openshift-tests-private]$ ls *rollback*
openshift-openshift-tests-private-release-4.12__amd64-rollback-nightly.yaml  openshift-openshift-tests-private-release-4.14__amd64-rollback-stable.yaml
openshift-openshift-tests-private-release-4.12__amd64-rollback-stable.yaml   openshift-openshift-tests-private-release-4.15__amd64-rollback-nightly.yaml
openshift-openshift-tests-private-release-4.13__amd64-rollback-nightly.yaml  openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml
openshift-openshift-tests-private-release-4.13__amd64-rollback-stable.yaml   openshift-openshift-tests-private-release-4.16__amd64-rollback-ec.yaml
openshift-openshift-tests-private-release-4.14__amd64-rollback-nightly.yaml  openshift-openshift-tests-private-release-4.16__amd64-rollback-nightly.yaml
```
We can add this job to there.

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly-4.15-upgrade-from-stable-4.15.yaml:59]: Thanks, done.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml [hunk 2]
```diff
@@ -101,6 +107,17 @@ tests:
     - ref: cucushift-chainupgrade-toimage
     - chain: openshift-upgrade-qe-sanity
     workflow: cucushift-installer-rehearse-aws-ipi-byo-route53
+- as: aws-ipi-ovn-hypershift-inplace-rollback-f7
+  cron: 2 15 1,8,15,22 * *
+  steps:
+    cluster_profile: aws-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      EXTRA_ARGS: --node-upgrade-type=InPlace
```
Comment (heliubj18) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml:116]: @shellyyang1989 Do you know why should we specify inplace upgrade type for the hosted cluster worker nodes in OTA-1232? The default upgrade type of worker nodes is replace.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml [hunk 3]
```diff
@@ -66,6 +66,12 @@ releases:
       architecture: amd64
       channel: candidate
       version: 4.15.0-rc.0
+  oldest-supported-z-rollback:
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml:69]: I do not think we need this release image setting, just like the existing OCP rollback job, we are using `latest` release image as the rollback target, that means, install with `release:latest` -> upgrade to `release:target` -> rollback to `release:latest`.

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml]: done

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml [hunk 4]
```diff
@@ -101,6 +101,17 @@ tests:
...
     - chain: openshift-upgrade-qe-sanity
     workflow: cucushift-installer-rehearse-aws-ipi-byo-route53
+- as: aws-ipi-ovn-hypershift-inplace-rollback-f7
+  cron: 2 15 1,8,15,22 * *
+  steps:
+    cluster_profile: aws-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      EXTRA_ARGS: --node-upgrade-type=InPlace
+      TEST_FILTERS: ~DisconnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&;~NonHyperShiftHOST&
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml:111]: For hypershift cluster, is it only run against a stable payload, right?

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-rollback-stable.yaml]: okay, move to *4.16__amd64-rollback-nightly.yaml

**Comment by shellyyang1989** [comment]
Hi Liangquan, thanks for the support. Please use replace profile instead. Thank you.

**Comment by LiangquanLi930** [comment]
> Hi Liangquan, thanks for the support. Please use replace profile instead. Thank you.

Done

**Comment by shellyyang1989** [comment]
The change looks good to me. Leaving it to @heliubj18 for review.

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-replace-rollback-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-replace-rollback-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-replace-rollback-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-replace-rollback-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-replace-rollback-f7

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-rollback-nightly.yaml [hunk 5]
```diff
@@ -27,6 +31,14 @@ base_images:
...
     namespace: ci
     tag: "4.16"
+  tests-private-postupg:
+    name: tests-private
+    namespace: ci
+    tag: "4.15"
+  tests-private-preupg:
+    name: tests-private
+    namespace: ci
+    tag: "4.15"
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-rollback-nightly.yaml:41]: Here should be 4.16, right?

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-inplace-rollback-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-inplace-rollback-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-inplace-rollback-f7

**Comment by LiangquanLi930** [comment]
@jianlinliu @shellyyang1989 PTAL, Thanks

File: ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+ref:
+  as: cucushift-downgrade-hypershift-worker-node-toimage
+  from: ansible
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml:3]: I guess we do not need to use `ansible` image here, because I did not see the step calls any ansible commands.

Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml]: okay, done, change to `upi-installer`

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml]: How about using verification-tests image to keep consistent with cucushift-upgrade-toimage step:
```
  from_image:
    namespace: ci
    name: verification-tests
    tag: latest
```

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml]: BTW, can you also update `cucushift-upgrade-hypershift-worker-node-toimage` together.

Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml]: https://github.com/openshift/verification-tests/blob/master/tools/openshift-ci/Dockerfile#L67
For the verification-tests image, it installs Ansible. I can confirm that we don't need to use the Ansible part.

Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml]: For hypershift upgrade and downgrade step, we primarily utilized OC commands, so using the UPI installer should cover it.

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml]: Or even to be cleaner, you can use `from: cli` for the hypershift upgrade/downgrade step.

Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml]: Thanks a lot, done

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/downgrade/hypershift/worker-node/toimage/cucushift-downgrade-hypershift-worker-node-toimage-ref.yaml]: Do you like to run a new pr-rehearse job to confirm the update of image still works.

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-inplace-rollback-f7

**Comment by LiangquanLi930** [comment]
@liangxia Could you help to review this PR?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-rollback-nightly.yaml [hunk 7]
```diff
@@ -87,6 +99,17 @@ tests:
           memory: 200Mi
     - ref: cucushift-chainupgrade-toimage
     workflow: cucushift-installer-rehearse-aws-ipi-byo-route53
+- as: aws-ipi-ovn-hypershift-inplace-rollback-f7
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-rollback-nightly.yaml:102]: The file name already contains `rollback`, so perhaps there is no need to add rollback to the test name, to avoid duplicate ? The job name is `periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-rollback-nightly-aws-ipi-ovn-hypershift-inplace-rollback-f7`

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-rollback-nightly.yaml]: Done, Thanks

**Comment by shellyyang1989** [comment]
Refers [OTA-1071](https://issues.redhat.com/browse/OTA-1071)
